### PR TITLE
Fix inconsistency between EventList.stack(), GTI.stack() and rest of stack methods.

### PR DESCRIPTION
**Pull request by LauraOlivera**
**Description**

This pull request fixes the existing inconsistency with the behavior of `EventList.stack()`, renaming it to `EventList.from_stack()`, which takes as an input a list of `EventList` objects. It also creates a new method` eventlist.stack(other)` to append a single `EventList` to an existing one.

### CODE REVIEW

File: gammapy/data/event_list.py [hunk 1]
```diff
@@ -93,6 +93,23 @@ def stack(cls, event_lists, **kwargs):
...
+        ----------
+        other : `~gammapy.data.EventList`
+            Event list to stack to self
+
+        Returns
+        -------
+        new_event_list : `~gammapy.data.EventList`
+            New event list
+        """
+        return self.__class__(vstack_tables([self.table, other.table]))
```
Comment (adonath) [inline gammapy/data/event_list.py:111]: On detail here: for the other classes we have such as `Map` or `Datasets`  the `.stack()` method work "in-place", this means it does not return anything but just modifies the existing data member. In this case you could just overwrite `self.table` and not return anything...

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @LauraOlivera, I have left one inline comment otherwise looks good to me!

File: gammapy/data/event_list.py (hunk 1 above)
Comment (LauraOlivera) [reply gammapy/data/event_list.py]: done. Initially I did it not in place because that is the way it is in `GTI.stack()`...

Comment (adonath) [reply gammapy/data/event_list.py]: Good to know! I'll check once again, but likely we are then going to change it for `GTI` as well...

Comment (LauraOlivera) [reply gammapy/data/event_list.py]: I can do that too, if you want

Comment (adonath) [reply gammapy/data/event_list.py]: Yes, please go ahead. Thanks!

**Review by adonath — APPROVED** [review]
Thanks a lot @LauraOlivera, I have no further comments...

**Pull request merged**